discovery_engine_core/web-api/src/personalization/routes.rs

```diff
@@ -235,6 +235,47 @@ async fn personalized_documents(
             return Err(NotEnoughInteractions.into());
         }
     }
+    let documents_by_interests = all_documents
+        .iter()
+        .enumerate()
+        .map(|(rank, document)| (document.id.clone(), rank))
+        .collect::<HashMap<_, _>>();
+
+    let categories = state.db.fetch_category_weights(&user_id).await?;
+    let mut documents_by_categories = all_documents
+        .iter()
+        .map(|document| {
+            let weight = document
+                .category
+                .as_deref()
+                .and_then(|category| categories.get(category).copied())
+                .unwrap_or_default();
+            (document.id.clone(), weight)
+        })
+        .collect_vec();
+    documents_by_categories.sort_unstable_by(|(_, a), (_, b)| a.cmp(b).reverse());
+    let documents_by_categories = documents_by_categories
+        .into_iter()
+        .enumerate()
+        .map(|(rank, (document_id, _))| (document_id, rank))
+        .collect::<HashMap<_, _>>();
+
+    let weight = state.config.personalization.rank_weight;
```

Comment:
:art:
I think this name is too generic. `interests_cat_bias` ? or any other name that is a bit more explicit about what this is.